projects/cm_mcu/commands/SensorControl.c

```diff
@@ -10,6 +10,272 @@
 #include "SensorControl.h"
 #include "common/smbus_helper.h"
 
+// Register definitions
+// -------------------------------------------------
+// 8 bit 2's complement signed int, valid from 0-80 C, LSB is 1 deg C
+// Same address for 4 XCVR and 12 Tx/Rx devices
+
+// two bytes, 12 FF to be disabled
+#define ECU0_14G_TX_DISABLE_REG 0x34U
+// one byte, 4 FF to be enabled/disabled (only 4 LSB are used)
+#define ECU0_25G_XVCR_TX_DISABLE_REG 0x56U
+// two bytes, 12 FF to be disabled
+#define ECU0_14G_RX_DISABLE_REG 0x34U
+// one byte, 4 FF to be enabled/disabled (only 4 LSB are used)
+#define ECU0_25G_XVCR_RX_DISABLE_REG 0x35U
+// one byte, 4 FF to be enabled/disabled (4 LSB are Rx, 4 LSB are Tx)
+#define ECU0_25G_XVCR_CDR_REG 0x62U
+// two bytes, 12 FF to be enabled/disabled. The byte layout
+// is a bit weird -- 0-3 on byte 4a, 4-11 on byte 4b
+#define ECU0_25G_TXRX_CDR_REG 0x4AU
+
+
+static int read_ff_register(const char *name, uint16_t packed_reg_addr, uint8_t *value, size_t size, int i2c_device)
+{
+  memset(value, 0, size);
+  // find the appropriate information for this FF device
+  int ff;
+  for (ff = 0; ff < NFIREFLIES; ++ff) {
+    if (strncmp(ff_moni2c_addrs[ff].name, name, 10) == 0)
+      break;
+  }
+  if (ff == NFIREFLIES) {
+    return -2; // no match found
+  }
+
+  int res;
+  if (i2c_device == I2C_DEVICE_F1){
+    while (xSemaphoreTake(ffldaq_f1_args.xSem, (TickType_t)10) == pdFALSE)
```

Comment:
suggest rewriting this like the following instead 
```
  SemaphoreHandle_t s = ffldaq_f1_args.xSem;
  if (i2c_device == I2C_DEVICE_F2) {
    t = ffldaq_f2_args.xSem;
  }
  while (xSemaphoreTake(s, (TickType_t)10) == pdFALSE)
    ;
```
Then you can replace lines around line 80 with just `xSemaphoreGive(s);`

Reply:
Okay, sounds good. Done.